src/bitcoin-cli.cpp

```diff
@@ -293,6 +302,13 @@ int CommandLineRPC(int argc, char *argv[])
             argc--;
             argv++;
         }
+        std::string rpcPass;
+        if (GetBoolArg("-stdinrpcpass", false)) {
+            if(!std::getline(std::cin,rpcPass)) {
+                std::cerr<<"error: -stdinrpcpass specified but failed to read from standard input\n";
+                return EXIT_FAILURE;
+            }
```

Comment:
Suggestion: `SetArg("-rpcpassword", rpcPass)` here, I think this avoids most of the other changes.